### DISCUSSION TITLE
At which point in the devcontainer's lifecycle is the dotfiles repository cloned?

### DISCUSSION THREAD

**Discussion by thevedicdeveloper** [devcontainers/spec#285] [Q&A] [answered]
When creating a development container for a repository to import our configurations into the new environment, at which point in the devcontainer's lifecycle does the dotfiles repository get cloned and the install.sh file executed?

**Comment by Anemia88** [comment]
Hacer efectiva la accion

**Comment by samruddhikhandale** [comment, marked as answer]
Hi 👋 

The dotfiles are cloned and installed after all of the "creation" phase lifecycle hooks, but before the `postStartCommand` and `postAttachCommand`. 

See https://github.com/devcontainers/cli/blob/main/src/spec-common/injectHeadless.ts#L375 for reference.
